Firestore/core/src/util/byte_stream_cpp.cc

```diff
@@ -54,8 +54,12 @@ StreamReadResult ByteStreamCpp::ToReadResult(const std::string& result) {
     input_->clear();
   }
 
+  // Saving the read count from reading `result`, because we are doing another
+  // read (peek) to determine if it is EOF yet.
+  auto result_read_count = input_->gcount();
+  bool is_eof = input_->eof() || (input_->peek() == EOF);
```

Comment:
Optional: the parentheses around the `EOF` comparison are not necessary (but fine to keep for readability).

Reply:
Same as above.